### ISSUE TITLE
Generate a secure random_string

### ISSUE THREAD

**Issue by TheAspens** [BOINC/boinc#2210]
In html/inc/util.inc there is a function called random_string().  This is implemented as

```
function random_string() {
      return md5(uniqid(rand(), true));
      return md5(uniqid(rand(), true));
 }
```
The PHP manuals warns for both rand() and uniqid():

> Caution.  This function does not generate cryptographically secure values, and should not be used for cryptographic purposes. 

Since random_string is used for the generation of various tokens including ones that are associated with identify, we should generate this string in a cryptographically secure way.

**Comment by knreed** [comment]
test

**Comment by AenBleidd** [comment]
This was fixed in 39cc8c05007fffc59c1a51a66511b7bd33f9642f
